x-pack/plugins/cloud_security_posture/public/pages/rules/rules_table.tsx

```diff
@@ -185,6 +184,29 @@ export const RulesTable = ({
   );
 };
 
+const EnableDisableRulesSwitch = ({
+  isRuleMuted: isRuleMutedProp,
+  changeCspRuleStateFn,
+}: {
+  isRuleMuted: boolean;
+  changeCspRuleStateFn: () => Promise<void>;
+}) => {
+  const [isRuleMuted, setIsRuleMuted] = useState<boolean>(isRuleMutedProp);
```

Comment:
This is an anti-pattern because we are setting the state for the component and then a refetch of the data will reset this again.